Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentMode.cpp

```diff
@@ -307,7 +307,15 @@ namespace PhysX
 
     AZStd::vector<AzToolsFramework::ViewportUi::ClusterId> JointsComponentMode::PopulateViewportUiImpl()
     {
-        return AZStd::vector<AzToolsFramework::ViewportUi::ClusterId>(m_modeSelectionClusterIds.begin(), m_modeSelectionClusterIds.end());
+        AZStd::vector<AzToolsFramework::ViewportUi::ClusterId> ids;
+        for (auto clusterid : m_modeSelectionClusterIds)
+        {
```

Comment:
Reserve the size in advance to avoid reallocations during additions?

`ids.reserve(m_modeSelectionClusterIds.size());`